crates/core/flags/complete/rg.zsh

```diff
@@ -434,7 +434,12 @@ _rg_types() {
   fi
 }
 
-_rg "$@"
+# don't run the completion function when being sourced by itself. See #2956
+if [[ "$funcstack[1]" = "_rg" ]] || ! type compdef >/dev/null 2>&1; then
```

Comment:
I originally wrote this as 

```zsh
if [[ "$funcstack[1]" = "_rg" ]]; then
  _rg "$@"
elif type compdef >/dev/null;
    compdef _rg rg
fi
```

But `ci/test-complete` didn't like that.  I feel like the test is wrong, because realistically you will either have `compdef` or will run the completion as `_rg`.  I stared at the test for a few seconds and didn't want to spend more time digging into it.

Reply:
Thanks! Done